### PR TITLE
disttask: fix subtask cancel bug

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #46258 

Problem Summary:
Dispatcher didn't count cancelled subtask, then mark task as success.
### What is changed and how it works?
CollectSubtaskError for cancelled subtask.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -381,11 +381,13 @@ func (stm *TaskManager) GetSubtaskInStatesCnt(taskID int64, states ...interface{
 
 // CollectSubTaskError collects the subtask error.
 func (stm *TaskManager) CollectSubTaskError(taskID int64) ([]error, error) {
-	rs, err := stm.executeSQLWithNewSession(stm.ctx, "select error from mysql.tidb_background_subtask where task_key = %? AND state = %?", taskID, proto.TaskStateFailed)
+	rs, err := stm.executeSQLWithNewSession(
```
Comment (D3Hunter) [inline disttask/framework/storage/task_table.go:384]: in which case will subtask be cancelled when the task state is not in(`canceling`, `reverting`)

Comment (ywqzzy) [reply disttask/framework/storage/task_table.go:384]: > in which case will subtask be cancelled when the task state is not in(`canceling`, `reverting`)

It is possible when using global sort, the 
```
minimalTasks, err := scheduler.SplitSubtask(ctx, subtask.Meta)
```
May return context.Cancel err.

**Review by wjhuang2016 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**